File: 643-maximum-average-subarray-i/maximum-average-subarray-i.java
class Solution {
    public double findMaxAverage(int[] nums, int k) {
        int curr=0;
        int n=nums.length;
        for(int i=0;i<k;i++){
            curr+=nums[i];
        }
        int max=curr;
        for(int i=1;i<=n-k;i++){
            curr=curr-nums[i-1]+nums[i+k-1];
            if(curr>max){
                max=curr;
            }
        }
        return (double)max/k;
        
    }
}

File: 643-maximum-average-subarray-i/maximum-average-subarray-i.py
class Solution(object):
    def findMaxAverage(self, nums, k):

        curr=0
        n=len(nums)
        for i in range(k):
            curr+=nums[i]
        maxx=curr
        for i in range (1,n-k+1):
            curr=curr-nums[i-1]+nums[i+k-1]
            if(curr>maxx):
                maxx=curr
        return maxx/float(k)
        """
        :type nums: List[int]
        :type k: int
        :rtype: float
        """
        